components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -276,6 +276,27 @@ private void preSetUserClaimOnMobileNumberUpdate(Map<String, String> claims, Use
         }
     }
 
+    /**
+     * Verify whether the mobile number update is invoked by the user himself, but not by another privileged user
+     * on behalf.
+     *
+     * @param user User whose claims are being updates.
+     * @return True if the user in the context is the same as the user whose claims are being updated, false otherwise.
+     */
+    private boolean isInvokedByUser(User user) {
+
+        String usernameFromContext = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        String tenantDomainFromContext = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        String userRealm = UserStoreConfigConstants.PRIMARY;
+        String[] strComponent = usernameFromContext.split("/");
```

Comment:
UserCoreUtil.extractDomainFromName
UserCoreUtil.removeDomainFromName

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/444/commits/f8d35a4708c69c10150800e4d79334d332bec0d0